Main/src/main/java/com/kusitms29/backendH/api/community/service/PostSearchService.java

```diff
@@ -0,0 +1,44 @@
+package com.kusitms29.backendH.api.community.service;
+
+
+import com.kusitms29.backendH.api.community.service.dto.response.PostSearchResponseDto;
+import com.kusitms29.backendH.domain.post.entity.Post;
+import com.kusitms29.backendH.domain.post.service.PostReader;
+import com.kusitms29.backendH.domain.user.entity.User;
+import com.kusitms29.backendH.domain.user.service.UserReader;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RequiredArgsConstructor
+@Transactional
+@Service
+public class PostSearchService {
+    private final UserReader userReader;
+    private final PostReader postReader;
+    public List<PostSearchResponseDto> searchPosts(Long userId, String keyword, Pageable pageable) {
+        User user = userReader.findByUserId(userId);
+
+        if(keyword == null || keyword.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        Page<Post> posts = postReader.searchByTitleOrContent(keyword, pageable);
+        return posts.stream()
+                .map(post -> PostSearchResponseDto.of(
+                        post.getId(), post.getPostType().getStringPostType(),
+                        post.getUser().getProfile(), post.getUser().getUserName(),
+                        post.getTitle(), post.getContent(), post.getCreatedAt()
+                ))
```

Comment:
1. [디미터의 법칙](https://dkswnkk.tistory.com/687)도 읽어보시면 좋을 것 같아요!
2. 가독성을 위한 개행이 필요해보여요!
```suggestion
                .map(post -> PostSearchResponseDto.of(
                        post.getId(), 
                        post.getPostType().getStringPostType(),
                        post.getUser().getProfile(), 
                        post.getUser().getUserName(),
                        post.getTitle(), 
                        post.getContent(), 
                        post.getCreatedAt(),
                ))
```
3. 정적 팩토리 메서드를 쓴다면 필요한 객체를 넘겨주세요! 필요한 모든 정보를 `post`가 알고 있으니 `post` 하나만 인자로 넘기고 메서드 안에서 작업하는 것도 좋아보여요!
```suggestion
                .map(PostSearchResponseDt::of)
```